src/ripple/ledger/tests/SkipList_test.cpp

```diff
@@ -0,0 +1,107 @@
+//-----------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/ledger/View.h>
+#include <ripple/basics/Log.h>
+#include <beast/unit_test/suite.h>
+
+namespace ripple {
+namespace test {
+
+class SkipList_test : public beast::unit_test::suite
+{
+    void
+    testSkipList()
+    {
+        beast::Journal const j;
+        std::vector<std::shared_ptr<Ledger>> history;
+        {
+            Config const config;
+            auto prev =
+                std::make_shared<Ledger>(create_genesis, config);
+            history.push_back(prev);
+            for (auto i = 0; i < 1023; ++i)
+            {
+                auto next = std::make_shared<Ledger>(
+                    open_ledger, *prev);
+                next->updateSkipList();
+                next->setClosed();
+                history.push_back(next);
+                prev = next;
+            }
+        }
+
+        {
+            auto l = *(std::next(std::begin(history)));
+            expect((*std::begin(history))->info().seq <
+                l->info().seq);
+            expect(hashOfSeq(*l, l->info().seq + 1,
+                j) == boost::none);
+            expect(hashOfSeq(*l, l->info().seq,
+                j) == l->info().hash);
+            expect(hashOfSeq(*l, l->info().seq - 1,
+                j) == l->info().parentHash);
+            expect(hashOfSeq(*history.back(),
+                l->info().seq, j) == boost::none);
+        }
+
+        // ledger skip lists store up to the previous 256 hashes
+        for (auto i = history.crbegin();
+            i != history.crend(); i += 256)
+        {
+            for (auto n = i;
+                n != std::next(i,
+                    (*i)->info().seq - 256 > 1 ? 257 : 256);
+                ++n)
+            {
+                expect(hashOfSeq(**i,
+                    (*n)->info().seq, j) ==
+                        (*n)->info().hash);
+            }
```

Comment:
The one thing I think that could improve this test is to check negative cases to ensure we're not incorrectly storing extra values in the skip list. It doesn't have to be exhaustive by any means.
Eg. You could check the edge case right here - after the `n` loop:
```
            expect(!hashOfSeq(**i,
                (*i)->info().seq - 258, j));
```
(I didn't worry about the underflow that'll occur on the last loop, because that also returns `none`.)